clang/test/Driver/nostdlibinc.c

```diff
@@ -1,10 +1,8 @@
-// RUN: %clang -target x86_64-unknown-unknown \
-// RUN:   -nostdlibinc -ffreestanding -fsyntax-only %s
+// RUN: %clang -target x86_64-unknown-unknown -nostdlibinc -ffreestanding -MM -MG %s | FileCheck -check-prefix=CHECK %s
```

Comment:
Could you please explain why we need to change community tests?

Does this mean we are changing existing behavior, and that users would need to update their flags/makefiles? Changing community tests could also cause merge conflicts when we upgrade.

Reply:
This test case is ensuring that the system `stdlib.h` cannot be found if `-nostdlibinc` is specified on the compilation commandline.
This can be ensured in more than one way - the approach chosen in this test case __assumes__ that `stdlib.h` is not present in `<build-dir>/lib/clang/11.0.0/include` directory.  This is an assumption because of the following reason:
The mechanism used by clang/llvm to add clang-specific declarations is to create a header file of the same name (ex.: `float.h`, `inttypes.h` - these are present in `<src-dir>/clang/lib/Headers`), use `#include_next <float.h>` to include the system header file, and add clang-specific declarations below this. Then they copy the header file from `<src-dir>/clang/lib/Headers` to `<build-dir>/lib/clang/11.0.0/include`. __The only reason that `stdlib.h`  is not present in `<build-dir/lib/clang/11.0.0/include` in the llvm upstream repo is because there has not been a need to add clang-specific declarations to `stdlib.h`.__  Had there been, this test case would have to be modified anyway.

Now, that we have added `stdlib.h` to `<build-dir>/lib/clang/11.0.0/include`, we need to change the way of ensuring that the "system" `stdlib.h` is not found when `-nostdlibinc` is specified on the compilation command line.

Reply:
Once we create a new macro for Checked C and use it as guard maybe we can simply add `-fno-checkedc-extension` to these tests to preserve community behavior without necessarily modifying the tests themselves.

Reply:
We should create a new macro for Checked C and use it as a guard.  But adding `-fno-checkedc-extension` to this test case will not stop it from failing. In it's original form, this test case will fail because of the fact that a file named `stdlib.h` exists in the `<build-dir>/lib/clang/11.0.0/include` directory, irrespective of what the file contains.

The above modification will be required to make this test case pass even if we create the new macro.